registration/registration-services/src/main/java/io/mosip/registration/validator/RequiredFieldValidator.java

```diff
@@ -25,65 +25,87 @@
 				@SuppressWarnings("rawtypes")
 				Map context = new HashMap();
 				context.put("identity", registrationDTO.getMVELDataContext());
 				VariableResolverFactory resolverFactory = new MapVariableResolverFactory(context);
 				required = MVEL.evalToBoolean(expression.get().getExpr(), resolverFactory);
 			}
-		}		
+		}
 		return required;
 	}
-	
-	
-	public List<String> isRequiredBiometricField(String subType, RegistrationDTO registrationDTO) throws RegBaseCheckedException {
+
+	@SuppressWarnings("unchecked")
+	public boolean isFieldVisible(UiSchemaDTO schemaField, RegistrationDTO registrationDTO)
+			throws RegBaseCheckedException {
+		boolean visible = true;
+		if (schemaField == null)
+			return visible;
+
+		if (schemaField.getVisible() != null && schemaField.getVisible().getEngine().equalsIgnoreCase("MVEL")
+				&& schemaField.getVisible().getExpr() != null) {
+
+			@SuppressWarnings("rawtypes")
+			Map context = new HashMap();
+			context.put("identity", registrationDTO.getMVELDataContext());
+			VariableResolverFactory resolverFactory = new MapVariableResolverFactory(context);
+			visible = MVEL.evalToBoolean(schemaField.getVisible().getExpr(), resolverFactory);
+		}
+		return visible;
+	}
+
+	public List<String> isRequiredBiometricField(String subType, RegistrationDTO registrationDTO)
+			throws RegBaseCheckedException {
 		List<String> requiredAttributes = new ArrayList<String>();
 		SchemaDto schema = identitySchemaService.getIdentitySchema(registrationDTO.getIdSchemaVersion());
-		List<UiSchemaDTO> fields = schema.getSchema().stream().filter(field -> field.getType() != null 
-				&& PacketManagerConstants.BIOMETRICS_DATATYPE.equals(field.getType()) 
-				&& field.getSubType() != null && field.getSubType().equals(subType)).collect(Collectors.toList());
-		
-		for(UiSchemaDTO schemaField : fields) {
-			if(isRequiredField(schemaField, registrationDTO) && schemaField.getBioAttributes() != null)
+		List<UiSchemaDTO> fields = schema.getSchema().stream()
+				.filter(field -> field.getType() != null
+						&& PacketManagerConstants.BIOMETRICS_DATATYPE.equals(field.getType())
+						&& field.getSubType() != null && field.getSubType().equals(subType))
+				.collect(Collectors.toList());
+
+		for (UiSchemaDTO schemaField : fields) {
+			if (isRequiredField(schemaField, registrationDTO) && schemaField.getBioAttributes() != null)
 				requiredAttributes.addAll(schemaField.getBioAttributes());
 		}
-		
-		//Reg-client will capture the face of Infant and send it in Packet as part of IndividualBiometrics CBEFF (If Face is captured for the country)
-		if ((registrationDTO.isChild()
-				&& APPLICANT_SUBTYPE.equals(subType) && requiredAttributes.contains("face"))
+
+		// Reg-client will capture the face of Infant and send it in Packet as part of
+		// IndividualBiometrics CBEFF (If Face is captured for the country)
+		if ((registrationDTO.isChild() && APPLICANT_SUBTYPE.equals(subType) && requiredAttributes.contains("face"))
```

Comment:
how do you decide child? is it configured on the server?

Reply:
Based on min age configuration sasi. age falling under min age is considered as child